src/Microsoft.AspNetCore.Routing/Tree/TreeRouter.cs

```diff
@@ -235,14 +235,11 @@ private struct TreeEnumerator : IEnumerator<UrlMatchingNode>
             private readonly Stack<UrlMatchingNode> _stack;
             private readonly PathTokenizer _tokenizer;
 
-            private int _segmentIndex;
```

Comment:
We don't need to keep a segment index because each UrlMatchingNode contains a Length property that indicates its depth inside the tree